### ISSUE TITLE
[MU4 Issue] [Workspaces] Workspace Name Character Limit

### ISSUE THREAD

**Issue by lengthwave** [musescore/MuseScore#11362]
**Describe the bug**
There is an ambiguous error when typing reeeeeeeeaaaaaallly long Workspace names. 
It could be better to give a fixed character limit prompt to the user instead of "cannot be used".

**To Reproduce**
1. Go to 'Workspaces'
2. Click on 'Create new workspace'
3. Create a really, really long name
4. See issue

Note: trying to test the limits of this, i did ignore the warning and was able to get it to accept it. Surprisingly, nothing broke with such a long name. 

**Expected behavior**
A warning that tells that workspace name should be less than X characters.

**Screenshots**
<img width="845" alt="Screen Shot 2022-04-25 at 6 54 19 PM" src="https://user-images.githubusercontent.com/30370609/165127646-647df4c9-6424-491b-84d0-57613e84437a.png">
<img width="773" alt="Screen Shot 2022-04-25 at 6 55 18 PM" src="https://user-images.githubusercontent.com/30370609/165127666-7e544881-cca5-4c85-ac07-b957980cbb25.png">
<img width="1083" alt="Screen Shot 2022-04-25 at 6 55 34 PM" src="https://user-images.githubusercontent.com/30370609/165127684-4849c313-47f0-4a78-a911-9a69e51c61ec.png">

**Platform information**
OS: macOS 12.2, Arch.: x86_64
MuseScore version (64-bit): 4.0.0-2214419458
revision: github-musescore-musescore-df96a36

**Comment by Tantacrul** [comment]
I suggest a cap of 24 characters. A user should not be able to enter more than that in the text field.

**Comment by Tantacrul** [comment]
@HemantAntony - wondering if you'd like this too?

**Comment by HemantAntony** [comment]
I'll take a look!

**Comment by poypoyan** [comment]
The workspace name is checked according to `mu::io::isAllowedFileName`  function in `\src/framework/global/io/path.cpp`. Also, this function is *only* used for this issue, checking the workspace name (in [here](https://github.com/musescore/MuseScore/blob/e8ea5b80809767f597e545806ceab9afb14f7cf7/src/workspace/view/newworkspacemodel.cpp#L104)). The relevant code there is:
```cpp
    // Check for trailing periods or spaces
    if (fn.back() == '.' || fn.back() == ' ') {
        return false;
    }

    // Check for pathnames that are too long
    if (fn.size() > 96) {
        return false;
    }
```
The error in the first image is due to trailing space, and not due to length. Of course the code also shows that the max length is actually 96. Hmmm... should we modify the path.cpp (the code here) or the newworkspacemodel.cpp (the link above)?

**Comment by DmitryArefiev** [comment]
Fixed in #14288
